keyboards/planck/keymaps/a2b/keymap.c

```diff
@@ -0,0 +1,866 @@
+ */
+[_TOUCHCURSOR] = LAYOUT_planck_grid(
+    KC_TAB,  KC_WH_U, KC_WH_L, KC_MS_U, KC_WH_R, KC_BTN1, KC_INS,  KC_HOME, KC_UP,   KC_PGUP, KC_BTN1, KC_ACL2,
+    KC_ESC,  GUI_WHD, ALT_MSL, KC_MS_D, CTR_MSR, KC_BTN2, KC_DEL,  KC_LEFT, KC_DOWN, KC_RGHT, KC_BTN2, KC_ACL1,
+    KC_CAPS, KC_ACL2, KC_ACL1, KC_ACL0, KC_BTN4, KC_BTN3, KC_BTN5, KC_END,  KC_DOWN, KC_PGDN, KC_BTN3, KC_ACL0,
+    _______, _______, _______, _______, _______, KC_SPC,  _______, _______, _______, _______, _______, _______
+),
+
+/* Adjust (Lower + Raise)
+ *,-----------------------------------------------------------------------------------------------------------.
+ *| Reset  | KeyClr |        |        |        |        |        | TermOn | TermOff|        |        | Del    |
+ *|--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
+ *| Esc    |        | MusMod | Aud On | Aud Off| AG Norm| AG Swap| Bépo   |        |        |        | Debug  |
+ *|--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
+ *|        | Voice- | Voice+ | Mus On | Mus Off| Midi On| MidiOff|        |        |        |        |        |
+ *|--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------|
+ *|        |        |        |        |        |        |        |        |        |        |        |        |
+ *`-----------------------------------------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_planck_grid(
+    RESET,   MVI_CLK, _______, _______, _______, _______, _______, TERM_ON, TERM_OFF,_______, _______, KC_DEL,
+    KC_ESC,  _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, BEPO,    AZERTY,  _______, _______, DEBUG,
+    _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+
+};
+
+uint16_t hextokeycode(int hex) {
+    if (hex == 0x0) {
+        return KC_P0;
+    } else if (hex < 0xA) {
+        return KC_P1 + (hex - 0x1);
+    } else {
+        return KC_A + (hex - 0xA);
+    }
+}
+
+void send_alt_code(uint16_t altcode)
+{
+  mod_state = get_mods(); // save state of mods and delete Shift and Alt
+  del_mods(MOD_MASK_SA);
+  register_code(KC_LALT);
+  for (int i = 12; i >= 0; i -= 4) {
+    tap_code(hextokeycode((altcode >> i) & 0xF));
+  }
+  unregister_code(KC_LALT);
+  set_mods(mod_state); // put back the mods state
+}
+
+#ifdef AUDIO_ENABLE
+  float plover_song[][2]     = SONG(PLOVER_SOUND);
+  float plover_gb_song[][2]  = SONG(PLOVER_GOODBYE_SOUND);
+#endif
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  if (IS_LAYER_ON(_AZERTY)) {
+    if (IS_LAYER_ON(_LOWER))  {
+      layer_on(_AZ_LOWER);
```

Comment:
You should NOT call layer functions from within layer_state_set_user, as it can lead to infinite loops. 

```suggestion
             state |= ((layer_state_t)1 << _AZ_LOWER);
```